lib/practice_exercises.rb

```diff
@@ -1,13 +1,24 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
```

Comment:
Since you are creating a new array the space complexity is O(n).  This is also **not** in place.